pkg/runtime/runtime.go

```diff
@@ -1330,7 +1330,7 @@ func (a *DaprRuntime) doProcessOneComponent(category ComponentCategory, comp com
 	case stateComponent:
 		return a.initState(comp)
 	}
-	return nil
```

Comment:
How is this error surfaced to the users? i.e do we:
- see an error in the log, or
- see a validation error when the component is created
- or see the controller crash (hopefully not)

Please document in the PR description

@artursouza @youngbupark @yaron2 is this a change we want to call out in the release note?

Reply:
The error is written to the log, I've updated the PR description.

Reply:
Thanks for updating the PR description.

Since the function only checks the `componentCategory`, I think it may be more natural to include a complain about the `componentCategory` here (it's fine to also keep component.Type if that brings more clarity).

In the error message, can you please also include the name of the violating component to make it easier for users to correct?

Reply:
I've rewritten the PR to capture the invalid `componentCategory` and include it in the error message.
The message written to the log now has the following text (which contains the component name and category):
```
component envvar-secret-store error, invalid component category: seccretstores
```